src/elf/program_header.rs

```diff
@@ -295,11 +297,10 @@ macro_rules! elf_program_header_std_impl { ($size:ty) => {
                 Ok(program_headers)
             }
 
-
             pub fn from_bytes(bytes: &[u8], phnum: usize) -> Vec<ProgramHeader> {
-                let bytes = unsafe { slice::from_raw_parts(bytes.as_ptr() as *mut ProgramHeader, phnum) };
-                let mut phdrs = Vec::with_capacity(phnum);
-                phdrs.extend_from_slice(bytes);
+                // FIXME: will panic if length of `bytes` is not equal to phnum * SIZEOF_PHDR.
+                let mut phdrs = vec![ProgramHeader::default(); phnum];
+                phdrs.as_mut_bytes().copy_from_slice(bytes);
```

Comment:
Can panic now. Originally, you could use this "safe" public function to read out of bounds data.
```
let b = [u8; 0];
let out_of_bounds = from_bytes(&b, 10000);
assert(out_of_bounds.len() == 10000);
```